_posts/2025-06-18-hibernate7-on-quarkus.adoc

```diff
@@ -0,0 +1,93 @@
+---
+layout: post
+title: 'Hibernate 7 on Quarkus: every new version is a better database experience for the users'
+date: 2025-06-18
+tags: hibernate, database, jpa
+synopsis: a roundup on the new upgrades of Hibernate on Quarkus
+author: lmolteni
+---
+:imagesdir: /assets/images/posts/hibernate7
+
+=== Introduction ===
+Hibernate is improving at a very fast speed, and so is its Quarkus support, as great database access is a key part of the Quarkus experience.
+The new Quarkus 3.24 release upgrades Hibernate to version 7, a major breaking change that requires some investigation with the https://docs.jboss.org/hibernate/orm/7.0/migration-guide/migration-guide.html[migration guide] when upgrading.
+Developers working on Hibernate and Quarkus are constantly collaborating, so here’s a quick peek at what has happened in the last months and what Quarkus users might expect in the future.
+
+=== License and Governance Updates ===
+Both Quarkus and Hibernate are now projects of the https://www.commonhaus.org[Commonhaus Foundation], a non-profit organization dedicated to the sustainability of open-source libraries.
+Since the upgrade to Hibernate 7, Quarkus and all modules of Hibernate now share the same open-source license: the https://www.apache.org/licenses/LICENSE-2.0[Apache License Version 2.0]. Implementing a proper open-source approach is crucial to the success of the two projects.
+
+=== Hibernate ORM 7.0 Updates ===
+The new version of Hibernate brings better performance and https://docs.jboss.org/hibernate/orm/7.0/whats-new/whats-new.html[all kinds of features], some of which improve the developer experience, such as https://docs.jboss.org/hibernate/orm/7.0/whats-new/whats-new.html#session-find-multiple[using findMultiple() and getMultiple()] to efficiently fetch entities in batches
+
+==== New Restrictions API ====
```

Comment:
If you move this below "Jakarta Data", you can mention that restrictions/orders can be used in Jakarta Data -- and perhaps provide an example, because that's pretty neat.
See https://docs.jboss.org/hibernate/orm/7.0/repositories/html_single/Hibernate_Data_Repositories.html#dynamic-restrictions